src/main/java/se/sundsvall/emailreader/integration/EWSMapper.java

```diff
@@ -0,0 +1,77 @@
+package se.sundsvall.emailreader.integration;
+
+
+import java.util.Base64;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import se.sundsvall.emailreader.api.model.Email;
+
+import microsoft.exchange.webservices.data.core.ExchangeService;
+import microsoft.exchange.webservices.data.core.PropertySet;
+import microsoft.exchange.webservices.data.core.enumeration.property.BasePropertySet;
+import microsoft.exchange.webservices.data.core.enumeration.property.BodyType;
+import microsoft.exchange.webservices.data.core.service.item.EmailMessage;
+import microsoft.exchange.webservices.data.core.service.schema.EmailMessageSchema;
+import microsoft.exchange.webservices.data.core.service.schema.ItemSchema;
+import microsoft.exchange.webservices.data.property.complex.EmailAddress;
+import microsoft.exchange.webservices.data.property.complex.FileAttachment;
+
+@Component
+public class EWSMapper {
+
+    private final Logger log = LoggerFactory.getLogger(EWSMapper.class);
+
+    private final PropertySet propertySetTextBody = new PropertySet(BasePropertySet.FirstClassProperties,
+        ItemSchema.Body, ItemSchema.Subject, ItemSchema.Attachments,
+        EmailMessageSchema.From, EmailMessageSchema.ToRecipients, ItemSchema.DisplayTo);
+
+    public EWSMapper() {
+        propertySetTextBody.setRequestedBodyType(BodyType.Text);
+    }
+
+    Email toEmail(final ExchangeService service, final EmailMessage emailMessage) throws Exception {
+
+        service.loadPropertiesForItems(List.of(emailMessage), propertySetTextBody);
```

Comment:
Is it possible to make the EWSMapper a non-component?

If this part is moved to EWSIntegration you will get less coupling. Maybe do the `service.loadPropertiesForItems` in EWSIntegration and send the result to this mapper?

Reply:
Done. A bit cleaner solution. Merci